integration/spark/src/main/spark3/java/io/openlineage/spark3/agent/lifecycle/plan/DataSourceV2RelationVisitor.java

```diff
@@ -89,6 +110,34 @@ private D findDatasetForIceberg(DataSourceV2Relation relation) {
             .build());
   }
 
+  private D findDatasetForAzureCosmos(DataSourceV2Relation relation) {
+    String relationName = relation.table().name().replace("com.azure.cosmos.spark.items.", "");
+    int expectedParts = 3;
+    String[] tableParts = relationName.split("\\.", expectedParts);
+    String tableName;
+    String namespace;
+    if (tableParts.length != expectedParts) {
+      tableName = relationName;
+      namespace = relationName;
```

Comment:
I kind of think the namespace should still be formatted as a URI if the table string isn't formatted as expected. Can the relation still work if the table isn't formatted as `service.database.collection`? If it can, how is the table name structured? Can we still construct an `azurecosmos://` namespace URI with the string we have?